native/packages/react-native-bpk-component-animate-height/src/BpkAnimateHeight.js

```diff
@@ -28,114 +28,118 @@ class BpkAnimateHeight extends React.Component {
     super(props);
 
     this.state = {
-      expanded: this.props.expanded,
       height: new Animated.Value(COLLAPSED_HEIGHT),
-      inTree: this.props.expanded,
-      expandAnimationInProgress: false,
     };
 
+    this.inTree = this.props.expanded;
+    this.expandAnimationInProgress = false;
+
     // overflow: hidden is not properly supported on Android,
     // which causes animation to look shoddy.
     // See https://facebook.github.io/react-native/releases/0.49/docs/layout-props.html#overflow
     this.animationDuration =
       Platform.OS === 'ios' ? props.animationDuration : 0;
-
-    this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
-    this.setExpandedHeight = this.setExpandedHeight.bind(this);
-    this.onHeightAnimationComplete = this.onHeightAnimationComplete.bind(this);
-    this.performAnimation = this.performAnimation.bind(this);
   }
 
   componentWillReceiveProps(nextProps) {
```

Comment:
@k0nserv told me that `componentWillReceiveProp` is marked for deprecation in React 16.3. Can we change it to `componentDidUpdate(prevProps)` instead?

Reply:
Perhaps in the future.